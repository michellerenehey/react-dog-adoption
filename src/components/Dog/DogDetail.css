.dog-detail {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #ffc0cb;
  border-radius: 5px;
  border: 1px black solid;
}
