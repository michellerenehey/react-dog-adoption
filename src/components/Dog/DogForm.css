.dog-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: solid 2px black;
}
