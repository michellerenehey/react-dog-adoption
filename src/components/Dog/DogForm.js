import React from 'react';

export default function DogForm({ name, setName, breed, setBreed, image, setImage, bio, setBio }) {
  return (
    <div>
      <form>
        <label>Dog Name:</label>
        <input
          placeholder="enter name"
          type="text"
          value={name}
          onChange={(e) => {
            setName(e.target.value);
          }}
        />
        <label>Dog Breed:</label>
        <input
          placeholder="enter breed"
          type="text"
          value={breed}
          onChange={(e) => {
            setBreed(e.target.value);
          }}
        />
        <label>Image Link:</label>
        <input
          placeholder="enter image url"
          type="text"
          value={image}
          onChange={(e) => {
            setImage(e.target.value);
          }}
        />
        <label>Dog Bio:</label>
        <textarea
          placeholder="enter breed"
          value={bio}
          onChange={(e) => {
            setBio(e.target.value);
          }}
        />
        <button>Update Doggy</button>
      </form>
    </div>
  );
}
