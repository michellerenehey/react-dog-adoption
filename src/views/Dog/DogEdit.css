.message {
  color: green;
  font-weight: bold;
  font-size: larger;
}
