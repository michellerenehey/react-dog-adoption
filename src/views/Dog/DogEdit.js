import { useState, useEffect } from 'react';
import { fetchDogById } from '../../services/dogs';

export default function DogEdit(props) {
  const id = props.match.params.id;
  const [name, setName] = useState('');
  const [breed, setBreed] = useState('');
  const [image, setImage] = useState('');
  const [bio, setBio] = useState('');
  console.log(id);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setName(data.name);
      setBreed(data.breed);
      setImage(data.image);
      setBio(data.bio);
      console.log(data);
    };
    fetchData();
  }, [id]);
  return <div>i am the dog edit component</div>;
}
