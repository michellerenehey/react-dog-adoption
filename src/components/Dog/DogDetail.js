import './DogDetail.css';
import { Link } from 'react-router-dom';

export default function DogDetail({ id, name, image, breed, bio, handleDelete }) {
  return (
    <div className="dog-detail">
      <h3>Meet {name}</h3>
      <img src={image} />
      <p>
        {name} is a(n) {breed}
      </p>
      <p>{bio}</p>
      <button>
        <Link to={`/dogs/${id}/edit`}>Edit</Link>
      </button>
      <button onClick={handleDelete}>Delete</button>
    </div>
  );
}
