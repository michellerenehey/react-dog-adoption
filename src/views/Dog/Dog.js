import { fetchDogById } from '../../services/dogs';
import { useEffect, useState } from 'react';

import DogDetail from '../../components/Dog/DogDetail';

export default function Dog(props) {
  const id = props.match.params.id;
  const [dog, setDog] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setDog(data);
      console.log(data);
    };
    fetchData();
  }, [id]);

  return <DogDetail {...dog} />;
}
