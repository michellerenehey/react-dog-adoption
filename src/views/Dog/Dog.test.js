import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Dog from './Dog';

test("should render a dog's details", async () => {
  const { container } = render(
    <MemoryRouter>
      <Dog match={{ params: { id: 3 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Meet Althea');
  expect(container).toMatchSnapshot();
});
