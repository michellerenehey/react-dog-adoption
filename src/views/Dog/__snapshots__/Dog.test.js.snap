// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a dog's details 1`] = `
<div>
  <div>
    <div
      class="dog-detail"
    >
      <h3>
        Meet 
        Althea
      </h3>
      <img
        src="https://placedog.net/500?id=10"
      />
      <p>
        Althea
         is a(n) 
        German Short-haired Pointer
      </p>
      <p>
        High energy pooch! Would do well in an active family with another dog to keep her company.
      </p>
    </div>
  </div>
</div>
`;
